src/pages/ItemList/Components/Filter.js

```diff
@@ -0,0 +1,109 @@
+import React, { useEffect } from 'react';
+
+import './Filter.scss';
+
+const initialValue = {
+  tealeaf: false,
+  pyramid: false,
+  teabag: false,
+  powder: false,
+  all: true,
+};
+
+const Filter = ({ whichProductRender, whatType, setWhatType, total }) => {
+  useEffect(() => {
+    setWhatType({ ...initialValue });
+  }, [whichProductRender]);
+
+  useEffect(() => {
+    if (
+      !whatType.tealeaf &&
+      !whatType.pyramid &&
+      !whatType.teabag &&
+      !whatType.powder
+    ) {
+      setWhatType({ ...initialValue });
+    }
+  }, [whatType.tealeaf, whatType.pyramid, whatType.teabag, whatType.powder]);
+
+  const handleAllButton = () => {
+    setWhatType({ ...initialValue });
+  };
+  const handleButton = i => {
+    setWhatType((whatType.all = false));
+    setWhatType({ ...whatType, [i]: !whatType[i] });
+  };
+  return (
+    <div className="filter">
+      <div>
+        <span>총</span>
+        <span className="itemNumber">{total.total_items}</span>
+        <span>개의 상품이 있습니다.</span>
+      </div>
+      <div className="buttons">
+        {whichProductRender.button === true && (
+          <>
+            <button
+              onClick={e => {
+                handleAllButton();
+              }}
+              id="all"
+              className={
+                whatType.all === true ? 'activatedButton' : 'unactivatedButton'
+              }
+            >
+              전체
+            </button>
+            <button
+              onClick={e => {
+                handleButton('tealeaf');
+              }}
```

Comment:
onClick 이벤트 핸들러로 전달되는 화살표 함수에 이벤트 객체를 매개변수로 받지 않아도 될 것 같습니다.
```suggestion
              onClick={() => {
                handleButton('tealeaf');
              }}
```